query/sources/common/src/main/java/org/codice/ddf/admin/sources/utils/RequestUtils.java

```diff
@@ -55,50 +56,46 @@
      * @return {@link ReportWithResultImpl} containing a {@link ResponseField}, or containing an {@link org.codice.ddf.admin.api.report.ErrorMessage}
      */
     public ReportWithResultImpl<ResponseField> sendGetRequest(UrlField requestUrl,
-            CredentialsField creds, Map<String, String> queryParams) {
+            CredentialsField creds, Map<String, Object> queryParams) {
         ReportWithResultImpl<ResponseField> responseResult = new ReportWithResultImpl<>();
         responseResult.addMessages(endpointIsReachable(requestUrl));
         if (responseResult.containsErrorMsgs()) {
             return responseResult;
         }
 
-        ReportWithResultImpl<Response> httpResponse = executeGetRequest(requestUrl,
-                creds,
-                queryParams);
-        if (httpResponse.containsErrorMsgs()) {
-            responseResult.addMessages(httpResponse);
-            return responseResult;
-        }
+        WebClient webClient = createWebClientBuilder(requestUrl.getValue(),
+                creds.username(),
+                creds.password(),
+                null).queryParams(queryParams)
+                .build();
 
-        return new ReportWithResultImpl<>(responseFieldFromResponse(httpResponse.result(),
-                requestUrl));
+        return sendGetRequest(webClient, requestUrl);
     }
 
     /**
-     * Executes a request by creating a Secure CXF Client from the provided url, credentials, and query params.
+     * Sends a GET request with the given {@code WebClient}
      * <p>
      * Possible Error Codes to return
      * - {@link org.codice.ddf.admin.common.report.message.DefaultMessages#CANNOT_CONNECT}
      *
-     * @param clientUrl   url to send GET request to
-     * @param creds       optional basic authentication
-     * @param queryParams additional query parameters
+     * @param webClient {@code WebClient} to send a GET request with
+     * @param urlField  the original request url
      * @return {@link Response} of the request
      */
-    public ReportWithResultImpl<Response> executeGetRequest(UrlField clientUrl,
-            CredentialsField creds, Map<String, String> queryParams) {
-        WebClient client = generateClient(clientUrl.getValue(), creds, queryParams);
-        ReportWithResultImpl<Response> report = new ReportWithResultImpl<>();
+    public ReportWithResultImpl<ResponseField> sendGetRequest(WebClient webClient,
+            UrlField urlField) {
+        ReportWithResultImpl<ResponseField> responseResult = new ReportWithResultImpl<>();
```

Comment:
We should be attempting to connect before sending the request in both POST and GET.